core/class_db.cpp

```diff
@@ -251,12 +251,18 @@ bool ClassDB::is_parent_class(const StringName &p_class, const StringName &p_inh
 		if (inherits == p_inherits) {
 			return true;
 		}
-		inherits = get_parent_class(inherits);
+		inherits = _get_parent_class(inherits);
 	}
 
 	return false;
 }
 
+bool ClassDB::is_parent_class(const StringName &p_class, const StringName &p_inherits) {
```

Comment:
You kept `OBJTYPE_RLOCK;` in `_is_parent_class`, so aren't you locking twice? It seems to contradict the comment in the header.

Reply:
Seems like you got it wrong in the `master` PR, but the `3.2` one is correct.

Reply:
Fixed.

Reply:
Wow good catch, merging is dangerous. Almost like the goto fail bug.